platform-validation/scripts/aziot-compatibility.sh

```diff
@@ -66,37 +69,46 @@ wrap_color() {
 }
 
 wrap_good() {
-    echo "$(wrap_color "$1" white): $(wrap_color "$2" green)"
+    if [ $VERBOSE -eq 1 ]; then
+        echo "$(wrap_color "$1" white): $(wrap_color "$2" green)"
+    fi
 }
 wrap_bad() {
-    echo "$(wrap_color "$1" bold): $(wrap_color "$2" bold red)"
+    echo "$(wrap_color "$1" bold red)"
+}
+
+wrap_debug_message() {
+
+    if [ $VERBOSE -eq 1 ]; then
+        echo "$(wrap_color "$1" white)"
+    fi
 }
 
-wrap_debug() {
-    echo "$(wrap_color "$1" white)"
+wrap_warning_message() {
+    echo "$(wrap_color "$1" magenta)"
 }
 
 wrap_pass() {
+    PASS=$((PASS + 1))
     echo "$(wrap_color "$1 - OK" green)"
 }
 wrap_fail() {
+    FAILURES=$((FAILURES + 1))
     echo "$(wrap_color "$1 - Error" bold red)"
 }
 
-wrap_warn() {
-    echo "$(wrap_color "$1 - Warning!!" yellow)"
-}
-
 wrap_warning() {
-    wrap_color >&2 "$*" magenta
+    WARNINGS=$((WARNINGS + 1))
+    echo "$(wrap_color "$1 - Warning!!" yellow)"
 }
 
 # ------------------------------------------------------------------------------
 #  Retrieve OS TYPE AND ARCHITECTURE (Required for Getting IoT)
 #  Derived from https://sh.rustup.rs
 # ------------------------------------------------------------------------------
 need_cmd() {
-    if ! check_cmd "$1"; then
+    ret=$($1 --help >/dev/null 2>&1)
```

Comment:
1. The methods described in the link mostly refer to bash. This is a sh script and I am trying to avoid any high level functions
2. > /dev/null 2>&1 means we are simply throwing the output and stderr away and only care about return code

Reply:
Oh woops, I misread that line